### ISSUE TITLE
Running deeptrio_dnm using deeptrio docker

### ISSUE THREAD

**Issue by karoliinas** [jguski/deeptrio_dnm#1]
Hi,

great paper and very interesting tool! I've used deepvariant and deeptrio using a docker container previously, but wondered how to incorporate the DNM/cDNM model into the analysis, if you had some example script of how to utilize your model in calling the variants in docker. Which deeptrio -version should I be using? Do I run all the steps (make_examples, call_variants and postprocess) with your model?

Also, does it output all the variants (for the trio) or just the de novos (for the child)? Does it take as long to run as deeptrio?

Thanks a million!

Karoliina

**Comment by jguski** [comment]
Hi Karoliina, 
thank you very much for your enquiry and your interest in our paper and tool. Unfortunately, I always used DeepTrio with Singularity but never with a Docker container, so I cannot give you a script for that. However, I think that it should be very similar to the use with Singularity, and there is a lot of documentation in the doc folder in this repository (https://github.com/jguski/deeptrio_dnm/tree/master/doc). I hope that you can find everything you need there, otherwise let me know. 
We developed the code with DeepTrio v1.3, and you can use the standard make_examples step. Of course you need to provide a custom truth VCF that labels the (c)DNMs accordingly as class 3 and 4. 
The Python scripts of our tool then operate on the examples created this way. 
The outputs of the calling.py script should generally be in line with DeepTrio, and will give you a VCF with the class probabilities for each variant (HOM/HET/ALT/DNM/cDNM). You can then filter it to keep only the denovos. Alternatively, it is possible to use the calling_dnm.py script to just get a table with denovos, which is also faster. All of this is described in the doc folder, too. 

I hope this helps you a bit. All the best for your project!

Jannis

**Comment by karoliinas** [comment]
Hi Jannis, thanks for the swift reply. Looking at the docs to call variants it seems I only need to run 2. and 4. (make_examples and call_variants). You also provide the example commands for these, which are very useful, thank you! 

However, I'm unsure how the deeptrio_dnm model is utilized, the example command looks like an ordinary deeptrio command. Where do I specify to use your model instead of the regular deeptrio model? Also about the truth vcf, would I need to call variants with multiple callers for each sample to generate the truth vcfs?

**Comment by jguski** [comment]
Hi Karoliina, yes, you only need to run 2. and 4. if you only want to call variants without any retraining. In that case, you actually don't even need to provide a truth VCF to the make_examples step. 
When you run calling.py, you can set the --model_path argument to the path of one of our pre-trained models (in the models folder of the repository). 
How exactly you get the labels for your variants is up to you. We used the overlap of DRAGEN and DeepVariant + some heuristics (see paper) to get the "ground truth" DNMs. Ideally, they would of course also be validated, e.g., via Sanger sequencing.
